CMakeLists.txt

```diff
@@ -1,27 +1,16 @@
-cmake_minimum_required(VERSION 3.2)
-project(trantor)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.2)
+PROJECT(trantor)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-#Clang, use c++17
-    set(DEFS "USE_STD_ANY")
-    set(USE_STD_ANY 1)
-    set(CMAKE_CXX_STD_FLAGS c++17)
-else()
-    EXEC_PROGRAM (gcc ARGS "--version | grep '^gcc'|awk '{print $3}' | sed s'/)//g' | sed s'/-.*//g'" OUTPUT_VARIABLE version)
-    MESSAGE(STATUS "This is gcc version:: " ${version})
-    if(version LESS 5.4.0)
-        MESSAGE(STATUS "gcc is too old")
-        stop()
-    elseif(version LESS 7.1.0)
-        set(CMAKE_CXX_STD_FLAGS c++14)
-        MESSAGE(STATUS "c++14")
-    else()
-        set(CMAKE_CXX_STD_FLAGS c++17)
-        set(DEFS "USE_STD_ANY")
-        set(USE_STD_ANY 1)
-        MESSAGE(STATUS "c++17")
-    endif()
-endif()
+IF(NOT DEFINED CMAKE_CXX_STANDARD)
+    SET(CMAKE_CXX_STANDARD 14)
+ELSE()
+    IF(CMAKE_CXX_STANDARD LESS 14)
+        MESSAGE(FATAL_ERROR "C++14 is required")
+    ENDIF()
+ENDIF()
```

Comment:
Lines 4 to 10 are still manually checking a global C++ version, I'd remove all these lines and down near where the project itself sets up its sources and links and so forth I'd add:
```cmake
set_property(TARGET trantor PROPERTY CXX_STANDARD 14)
```
This will mandate C++14 for your project, even if a user project requires higher.

In addition you could always set a default for the global `CMAKE_CXX_STANDARD` if it doesn't exist, but I prefer to leave that up to the users as a library should not be overriding their choices.

Reply:
@OvermindDL1  I removed the lines and use `set_property(TARGET trantor PROPERTY CXX_STANDARD 14)`, but for test targets, I have to add a long list to the set_property command, Is there an easier way to do this?
https://github.com/an-tao/trantor/blob/2eb8efb56543a5bbaf0ae8bee6f907c19cf86a7e/trantor/tests/CMakeLists.txt#L38-L55